packages/playwright/src/plugins/plugin.ts

```diff
@@ -0,0 +1,254 @@
+import { readdirSync } from 'fs';
+import { readTargetDefaultsForTarget } from 'nx/src/project-graph/utils/project-configuration-utils';
+import { basename, dirname, join, relative } from 'path';
+
+import {
+  CreateNodes,
+  CreateNodesContext,
+  joinPathFragments,
+  TargetConfiguration,
+} from '@nx/devkit';
+import { getNamedInputs } from '@nx/devkit/src/utils/get-named-inputs';
+
+import type { PlaywrightTestConfig } from '@playwright/test';
+import { getFilesInDirectoryUsingContext } from 'nx/src/utils/workspace-context';
+import minimatch = require('minimatch');
+import { loadPlaywrightConfig } from '../utils/load-config-file';
+
+export interface PlaywrightPluginOptions {
+  targetName?: string;
+  ciTargetName?: string;
+}
+
+interface NormalizedOptions {
+  targetName: string;
+  ciTargetName?: string;
+}
+
+export const createNodes: CreateNodes<PlaywrightPluginOptions> = [
+  '**/playwright.config.{js,ts,cjs,cts,mjs,mts}',
+  async (configFilePath, options, context) => {
+    const projectRoot = dirname(configFilePath);
+
+    // Do not create a project if package.json and project.json isn't there.
+    const siblingFiles = readdirSync(projectRoot);
+    if (
+      !siblingFiles.includes('package.json') &&
+      !siblingFiles.includes('project.json')
+    ) {
+      return {};
+    }
+
+    const normalizedOptions = normalizeOptions(options);
+    const projectName = basename(projectRoot);
+
+    return {
+      projects: {
+        [projectName]: {
+          root: projectRoot,
+          projectType: 'library',
+          targets: await buildPlaywrightTargets(
+            configFilePath,
+            projectRoot,
+            normalizedOptions,
+            context
+          ),
+        },
+      },
```

Comment:
```suggestion
      projects: {
        [projectRoot]: {
          root: projectRoot,
          targets: await buildPlaywrightTargets(
            configFilePath,
            projectRoot,
            normalizedOptions,
            context
          ),
        },
      },
```

we don't want the name or the type, iirc!